funtofem/model/function.py

```diff
@@ -170,7 +173,8 @@ def optimize(
         self.optim = True
         self.lower = lower
         self.upper = upper
-        self.scale = scale
+        if scale is not None:
```

Comment:
Do you have a catch for `scale is None` that I'm not seeing? Do other parts of the code depend on `self.scale` not being `None`?

Reply:
Done